### ISSUE TITLE
Pagination on table_for inside sections and panels

### ISSUE THREAD

**Issue by ghost** [activeadmin/activeadmin#1116]
Thanks everyone for AA, this is absolutely wicked!

Would love to see a way to implement a DRY pagination to table_for inside sections and panels on both the dashboard and in model show pages when displaying its children items

Sorry if this is already out there.

**Comment by lanrion** [comment]
@jdmyid me, too.

**Comment by sunloverz** [comment]
guys, using will_paginate and it doesnt work, how could I handle that issue?

**Comment by abrambailey** [comment]
I have the Kaminari initializer setup appropriately and using the following code. Getting stack error:
```ruby
      users = role.users
      panel 'Users' do
        paginated_collection(users.page(params[:users_page]).per_page_kaminari(15), download_links: false) do
          table_for(collection, sortable: false) do
            column :id
           end
         end
       end
```

**Comment by filipkis** [comment]
Just confirming, this is still an issue. `.per` throws undefined method error and `.per_page_kaminari` ends in stack overflow.

**Comment by filipkis** [comment]
OK, with some debugging I realized what is the issue. It's the combination of `.page` and `.per_page_kaminari` that's causing the problem. If you have `will_paginate` gem, it's likely you renamed the Kaminari method from `page` to `per_page_kaminari`, so calling both causes an issue as first triggers the `will_paginate` and second then fails as it calls some methods (namely `first`) that get executed by `will_paginate` and result in stack overflow. 

So, if you're getting undefined error when calling it with just `per` (as in [original answer](https://github.com/activeadmin/activeadmin/issues/1116#issuecomment-4615857)

```ruby
    panel :test do
      paginated_collection(my_collection.per_page_kaminari(params[:page]).per(15), download_links: false) do
        table_for(collection, sortable: false) do
          column :start
          column :stop
          column :duration        
        end
      end
    end
```

In other words, it's not the `per` method that needs to be replaced, but the `page` method.

**Comment by tordans** [comment]
Just in case someone stumbles on this thread, this is what worked for me:

Szenario: Active admin view with **multiple, separately paginated lists**.
`params[:somethings_page]` and `param_name: :somethings_page` need to be unique per panel.

Gemfile: `gem 'kaminari'`

```ruby
column do
  panel "PanelName" do
    paginated_collection(user.something_scope.page(params[:somethings_page]).per(10), param_name: :somethings_page, download_links: false) do
      table_for(collection) do
        column 'Titel' do |something|
          link_to(something.title, backoffice_something_path(something))
        end
      end
    end
  end
end
```

No additional code was needed.
